src/io/pti.jl

```diff
@@ -943,4 +943,744 @@ function _populate_defaults!(data::Dict)
+        # Get Dict in a PSSE way
+        psse_comp = _pm2psse_switched_shunt(shunt)
+
+        # Print it in the file
+        _print_pti_str(io, psse_comp, _pti_dtypes["SWITCHED SHUNT"])
+    end
+
+    println(io, "0 /END OF SWITCHED SHUNT DATA, BEGIN GNE DEVICE DATA")
+
+    Memento.warn(_LOGGER, string("Export GNE data is not yet supported"))
+
+    println(io, "0 /END OF GNE DEVICE DATA")
+    println(io, "Q")
+end
+
+
+"Given a PTI component dict print it in the .raw file"
+function _print_pti_str(io::IO, component, _dtype)
+    str = ""
+    for (data, type) in _dtype
+        if type == String
+            # Avoid losing the trailing whitespace in ID of components
+            str *= "$(component[data]),"
+        else
+            str *= "$(component[data]),\t"
+        end
+    end
+    println(io, str)
+end
+
+
+"""
+Export remaining keys of a componet
+"""
+function _export_remaining!(sub_data::Dict{String, Any}, pm_comp::Dict{String, Any}, pti_default::Dict{String, Any})
+    # Try to get the value from the PM component (is in lower case), else use the default
+    for (key, default) in pti_default
+        if ! haskey(sub_data, key)
+            sub_data[key] = get(pm_comp, lowercase(key), default)
+        end
+    end
+end
+
+
+"""
+Create a header for the case
+"""
+function _pm2psse_header(pm::Dict{String, Any})
+    sub_data = Dict{String, Any}()
+
+    for (dtype, _) in _transaction_dtypes
+        sub_data[dtype] = get(pm, dtype, _default_case_identification[dtype])
+    end
+
+    return sub_data
+end
+
+
+"""
+Parses PM Bus data to PSS(R)E-style.
+"""
+function _pm2psse_bus(pm_bus::Dict{String, Any})
+    sub_data = Dict{String, Any}()         
+    sub_data["I"] = pm_bus["bus_i"]
+
+    if haskey(pm_bus, "name")
+        sub_data["NAME"] = "\'$(pm_bus["name"])\'"
+    elseif haskey(pm_bus, "string")
+        sub_data["NAME"] = "\'$(pm_bus["string"])\'"
+    else
+        sub_data["NAME"] = "\'$(_default_bus["NAME"])\'"
+    end
+
+    sub_data["BASKV"] = get(pm_bus, "base_kv", _default_bus["BASKV"])
+    sub_data["IDE"] =  get(pm_bus, "bus_type", _default_bus["IDE"])
+    sub_data["AREA"] = get(pm_bus, "area", _default_bus["AREA"])
+    sub_data["ZONE"] = get(pm_bus, "zone", _default_bus["ZONE"])
+    sub_data["OWNER"] = get(pm_bus, "owner", _default_bus["OWNER"])
+    sub_data["VM"] = get(pm_bus, "vm", _default_bus["VM"])
+    sub_data["VA"] = get(pm_bus, "va", _default_bus["VA"])
+    sub_data["NVHI"] = get(pm_bus, "vmax", _default_bus["NVHI"])
+    sub_data["NVLO"] = get(pm_bus, "vmin", _default_bus["NVLO"])
+
+    _export_remaining!(sub_data, pm_bus, _pti_defaults["BUS"])
+    
+    return sub_data
+end
+
+
+"""
+Parses PM Load data to PSS(R)E-style.
+"""
+function _pm2psse_load(pm_load::Dict{String, Any}, area::Int64, owner::Int64, zone::Int64)
+    
+    sub_data = Dict{String, Any}()
+    sub_data["I"] = pm_load["load_bus"] 
+    sub_data["ID"] = "\'$(pm_load["source_id"][end])\'" # CHECK WITH DICTS FROM MATPOWER
```

Comment:
"source_id" is an optional parameter, this should include a default value as well.  I think the empty string in single quotes is valid for raw files.

Reply:
I prefer to use the default value or using the ID to enumerate loads at a particular bus to avoid problems of loads with same bus and ID in PSSE